### PR TITLE
[SOB-66] Jwt auth

### PR DESCRIPTION
**Pull request by Andmedoctopus**

### CODE REVIEW

**Comment by Andmedoctopus** [comment]
COOL

**Comment by Andmedoctopus** [comment]
> COOL

NOT cool

File: src/sobesity/jwt.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if auth_header is None:
+        response = jsonify({"message":"No token provided"})
+        response.status_code = 401
+        abort(response)
+
+    token = auth_header.split(" ")[1]
+    try:
+        get_user_id_from_jwt(token)
+    except Exception as exc:
+        breakpoint()
```
Comment (Andmedoctopus) [inline src/sobesity/jwt.py]: Remove breakpoint

File: src/sobesity/jwt.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        pass
+
+def get_user_id_from_jwt(token)-> UserId:
+    try:
+        payload = jwt.decode(token, "secret", algorithms=["HS256"])
+    except jwt.exceptions.InvalidSignatureError:
+        raise
+    return payload["sub"]
```
Comment (Andmedoctopus) [inline src/sobesity/jwt.py]: Remove magic string

File: src/sobesity/jwt.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+def encode_jwt(user_id: UserId) -> str:
+    return jwt.encode({"sub": user_id}, "secret", algorithm="HS256")
```
Comment (Andmedoctopus) [inline src/sobesity/jwt.py]: Add user jwt struct

**Review by Zhental — DISMISSED** [review]

**Review by vorobeybird — CHANGES_REQUESTED** [review]
Please add 
1) current_user endpoint
2) Name and Surname fields to api/user endpoint 
image

**Comment by Andmedoctopus** [comment]
> Please add
> 
>     1. current_user endpoint
> 
>     2. Name and Surname fields to api/user endpoint
>        image

Added

**Review by vorobeybird — APPROVED** [review]

**Pull request merged**